client/my-sites/plugins/plugin-details-header/index.jsx

```diff
@@ -28,7 +28,9 @@ const PluginDetailsHeader = ( { plugin, isPlaceholder } ) => {
 					<div className="plugin-details-header__info">
 						<div className="plugin-details-header__info-title">{ translate( 'Developer' ) }</div>
 						<div className="plugin-details-header__info-value">
-							<a href={ plugin.author_url }>{ plugin.author_name }</a>
+							<a href={ `/plugins?s=author:"${ plugin.author_name || '' }"` }>
+								{ plugin.author_name }
+							</a>
```

Comment:
Trying this for Contact form returns `author:"Takayuki Miyoshi".` which returns no results. Using `takayukister`  on the other hand returns results where `takayukister` is a contributor.

```
author: "<a href="1reh-p2">Takayuki Miyoshi</a>",
author_profile: "https://profiles.wordpress.org/takayukister/",
```

Reply:
Agreed, applied your suggestion to look for `author_profile` at https://github.com/Automattic/wp-calypso/pull/60977/commits/01bc0de5db1de6776476d10e395564fd738f6147.

---

But for yoast for example the author profile is `joostdevalk`, which will return the correct values on the WP.org request, but won't on the WP.com side. What are your opinions on this case?

Reply:
We can add that information as plain text to the `author_profile: joostdevalk ` in WP.com and search in both the Author and the Author profile?

Reply:
BTW, I think we still have a problem with WP.org too

<img width="870" alt="CleanShot 2022-02-16 at 10 07 40@2x" src="https://user-images.githubusercontent.com/12430020/154222151-38164f92-b9c1-4044-8bdf-78fb0d14e919.png">

<img width="867" alt="CleanShot 2022-02-16 at 10 28 41@2x" src="https://user-images.githubusercontent.com/12430020/154225273-776f37de-f712-41b3-bba0-5f4c3bf4bca1.png">

<img width="855" alt="CleanShot 2022-02-16 at 10 29 34@2x" src="https://user-images.githubusercontent.com/12430020/154225405-35e6c395-81ba-49b3-ac3a-d1db5741e840.png">


I think we need to search by the first contributor? To recap:

- `author` contains a "display" name that doesn't match searches
- `author_profile` contains an author slug in the last part, but might the contributor who published it?
- `contributors` seems to contain all contributors in an ordered list. Perhaps by number of contributions.

Reply:
Again, contributors is faulty

<img width="853" alt="CleanShot 2022-02-16 at 10 35 43@2x" src="https://user-images.githubusercontent.com/12430020/154226362-e762a9b0-40e8-439f-a998-59cf4a93737d.png">

the `author` has the actual author but not a `slug` we can use for searching...

Here is an idea, loop through `contributors` and check which contributor `display_name` is contained in the `author`.

example `https://api.wordpress.org/plugins/info/1.2/?action=plugin_information&request%5Bslug%5D=clicky`

<img width="862" alt="CleanShot 2022-02-16 at 10 40 16@2x" src="https://user-images.githubusercontent.com/12430020/154227064-80ce1d7d-8740-44f7-8e2a-1e14821ba7bd.png">

`display_name: "Yoast"` is contained to `author: "<a href="https://yoast.com/">Team Yoast</a>"` so the author would be `yoast`.

Reply:
Yes, I'll try this option, just keep in mind it can be tricky in some cases as *Woocommerce Amazon Pay** that will return the author keyword as `woocommerce` instead of automattic.

source: https://user-images.githubusercontent.com/12430020/154225405-35e6c395-81ba-49b3-ac3a-d1db5741e840.png

Reply:
Change implemented at https://github.com/Automattic/wp-calypso/pull/60977/commits/8a9f3bf36eab115c7ece61ea65304a72330f60eb

And beside some cases with woocommerce, woothemes, and automattic (those seem to be together as authors of lot of plugins) it seems to be working pretty well.

Reply:
While testing using the WooCommerce Subscription plugin I have found that clicking on `Automattic` filters by `woocommerce` author. This is currently expected because the `author_profile` is set to `woocommerce` so the filter applies that keyword. Please see screenshots: 
<img width="776" alt="Screenshot 2022-02-18 at 09 45 40" src="https://user-images.githubusercontent.com/3519124/154654950-9486498c-10fb-4624-9849-5809675ec78b.png">
<img width="518" alt="Screenshot 2022-02-18 at 09 48 16" src="https://user-images.githubusercontent.com/3519124/154654977-0971e81b-75e7-43ec-b7f1-bffa381cc7a9.png">

But that was unexpected for me as a user, I was expecting filtering by `Automattic` as displayed in the link. 

Changing the priority to use `author_name` will conflict with other filters like filtering by `Takayuki Miyoshi` as described above so maybe there is a better criteria we can find? Sorry for leaving an open question but I thought is worth discussing it.
<img width="572" alt="image" src="https://user-images.githubusercontent.com/3519124/154656403-8f618265-1605-4ffa-973b-3e67f7199b4b.png">

Reply:
Yes, you got it right, the displayed name cannot be used for the search because the WP.org request expects the contributor id. 

That's why we try to relate the contributor name with the displayed name to retrieve the keyword, which will work in most cases.

But on WP.com plugins (paid plugins) we currently don't use the contributors' property, that's why it may pick the author profile which can be "unrelated" to the displayed name.